File: backend/src/main/java/com/artifax_solutions/website/model/PortfolioShowcase.java
package com.artifax_solutions.website.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class PortfolioShowcase {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String subTitle;
    private String title;
    private String description;

    @OneToMany(mappedBy = "portfolioShowcase")
    private List<PortfolioShowcaseCard> showCaseCardList;
}

File: frontend/src/interface/PortfolioShowcase.ts
export interface PortfolioShowcase {
  id: number | null | undefined;
  subTitle: string | null | undefined;
  title: string | null | undefined;
  description: string | null | undefined;
  imageUrl?: string | undefined | null;
  filterOptions?: string[] | undefined | null;
  showCaseCardList: PortfolioShowcaseCard[] | undefined | null;
}
export interface PortfolioShowcaseCard {
  id: number | null | undefined;
  imageUrl: string | null | undefined;
  altText: string | null | undefined;
  title: string | null | undefined;
  description: string | null | undefined;
}
